### PR TITLE
Generic overall fixes

### PR DESCRIPTION
**Pull request by guilhermeblanco**

### CODE REVIEW

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php [hunk 1]
```diff
@@ -412,7 +411,7 @@ class ClassMetadataInfo implements ClassMetadata
      *
      * @var array
      *
-     * @todo We could get rid of this array by just using $fieldMappings[$fieldName]['columnName'].
+     * @todo 3.0 Remove this.
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php]: Please add a `@deprecated`

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php [hunk 2]
```diff
@@ -1569,13 +1581,16 @@ protected function _validateAndCompleteOneToOneMapping(array $mapping)
         if ($mapping['isOwningSide']) {
             if ( ! isset($mapping['joinColumns']) || ! $mapping['joinColumns']) {
                 // Apply default join column
-                $mapping['joinColumns'] = array(array(
-                    'name' => $this->namingStrategy->joinColumnName($mapping['fieldName'], $this->name),
-                    'referencedColumnName' => $this->namingStrategy->referenceColumnName()
-                ));
+                $mapping['joinColumns'] = array(
+                    array(
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php]: Previous syntax was ok here

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php [hunk 3]
```diff
@@ -1689,16 +1707,23 @@ protected function _validateAndCompleteManyToManyMapping(array $mapping)
                 && (! (isset($mapping['joinTable']['joinColumns']) || isset($mapping['joinTable']['inverseJoinColumns'])));
 
             if ( ! isset($mapping['joinTable']['joinColumns'])) {
-                $mapping['joinTable']['joinColumns'] = array(array(
+                $mapping['joinTable']['joinColumns'] = array(
+                    array(
                         'name' => $this->namingStrategy->joinKeyColumnName($mapping['sourceEntity'], $selfReferencingEntityWithoutJoinColumns ? 'source' : null),
                         'referencedColumnName' => $this->namingStrategy->referenceColumnName(),
-                        'onDelete' => 'CASCADE'));
+                        'onDelete' => 'CASCADE'
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php]: Previous syntax was ok, except for closing `)` and trailing comma

File: tests/Doctrine/Tests/OrmTestCase.php [hunk 4]
```diff
@@ -57,40 +59,41 @@
      */
     protected function createAnnotationDriver($paths = array(), $alias = null)
     {
-        if (version_compare(\Doctrine\Common\Version::VERSION, '3.0.0', '>=')) {
-            $reader = new \Doctrine\Common\Annotations\CachedReader(
-                new \Doctrine\Common\Annotations\AnnotationReader(), new ArrayCache()
-            );
-        }
-        else if (version_compare(\Doctrine\Common\Version::VERSION, '2.2.0-DEV', '>=')) {
+        if (version_compare(Version::VERSION, '3.0.0', '>=')) {
```
Comment (Ocramius) [inline tests/Doctrine/Tests/OrmTestCase.php]: Do we even have `3.0`? O_o

Comment (guilhermeblanco) [inline tests/Doctrine/Tests/OrmTestCase.php]: You ask me? =)

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php (hunk 3 above)
Comment (guilhermeblanco) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php]: Nah... unreadable.

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php (hunk 2 above)
Comment (guilhermeblanco) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php]: Nah... unreadable.

**Pull request merged**